### ISSUE TITLE
Document current HA model (Epic)

### ISSUE THREAD

**Issue by l-mb** [s3gw-tech/s3gw#361]
### What needs to be done

Document the high-availability guarantees of s3gw on top of Longhorn.

- What is the expected fail-over time in response to a node/process failure
- What does it depend on (LH setup, config options, ingress used ...)

### Why it needs to be done

Users of storage solutions expect to understand the HA model of their solutions to gauge whether it's an appropriate choice for their workloads.

This is a first step in identifying what s3gw has as-is. What's the reasonable/best thing we can do today?

Once we have that, we should have follow ups on how to improve it (based on user/PM requirements) to enable further use cases. We should, once we have this documented, also consider if we can test/validate these claims.

### Acceptance Criteria

Blog post, possibly part of the project's official documentation.

This could initially be implemented similarly to Longhorn or Rancher Telemetry. Perhaps there is an opportunity for code sharing.

### Tasks
- [ ] #685
- [ ] #733
- [ ] #803
- [ ] #841

**Comment by giubacc** [comment]
Is there an idea/expectation when to allocate this activity in the context of LH integration?
The topic seems quite vast (and important).

**Comment by jhmarina** [comment]
Yes, @giubacc this is planned for Milestone 4 of the Experimental Schedule, which is our v0.22.0 milestone.

**Comment by jecluis** [comment]
@giubacc will be doing some initial research into what we can get out of the underlying system, and its limitations, before we evaluate what are our options to move this forward.

**Comment by giubacc** [comment]
what still needs to be done:

- testing the s3gw when there is an actual node fault (not the taint but a more drastic `poweroff -f`)
- integrate a client in the restart loop and measure how many ops are failing/succeding/hanging
- finalize the research PR
- make an ADR simply pointing the research and decretating the HA model for the s3gw.

other?

**Comment by jecluis** [comment]
@giubacc if there are pending tasks in this Epic (which I'm sure there are, at least with regard to documentation), please add them as tasks to this issue.

**Comment by giubacc** [comment]
There is a on-going task (medik8s research) regarding the HA, I think we will have to wait until the definitive path for HA will be set, I'd keep this opened because we will have to write something in the documentation at some point, but now we still don't know exactly what

**Comment by jecluis** [comment]
@giubacc please file an issue for the documentation, and whatever you think may be needed to track the effort, and add it to this epic.
